### DISCUSSION TITLE
Automatically update Albums/unsorted view when pictures are added to an album

### DISCUSSION THREAD

**Discussion by bitelxux** [photoprism/photoprism#2713] [Ideas]
The same way when a selection of photos is automatically removed from the general view when they are archived, it would be of a great  help to organize photos in albums if they where automatically removed from the Albums/unsorted view when they are added to an album.

Current behavior makes difficult to track what pictures have already been moved to an album and the only way to remove them from the view is to refresh the page, what resets to the beginning.

Hopefully a pretty straight forward change.

**Comment by lastzero** [comment]
I don't quite understand it yet... when you add a photo to an album again, it doesn't show up multiple times - the action is just ignored. Could you provide screenshots showing where the problem is and/or what should be changed?

**Comment by bitelxux** [comment]
Hi, the scenario is as follows

I have potentially thousands of pictures (pics in no album) to organize in albums
The Albums/unsorted view helps me showing only those pictures that are not
in any album.
Then I start selecting pictures and adding them to albums

During the process, when I add a selection to an album, all the selection
marks in the albums/unsorted view are cleared but the photos are still
shown until the view albums/unsorted is refreshed. That makes pretty hard
to keep track of what pictures have been organized in the current
'organizing session'

Ideally, when we are in the albums/unsorted view, pictures that are added
to an album should automatically be removed from the view without the need
of refreshing. Or, if you like, the albums/unsorted view should be
automatically updated.

I hope it makes it clearer now.

**Reply by lastzero** [reply]
I understand. This could, in the worst case, cause certain states of the user interface to be lost, such as the scroll position, or if you happen to be editing something inline in the view at the same time, the input value would be lost. Therefore, we are usually very careful with reloads, because if we do it, it needs to be tested very well. Note that our team is small and our time is very limited due to the many feature requests.

**Reply by lastzero** [reply]
Feel free to open a feature request/idea on GitHub Issues describing the issue similar to what you just wrote (i.e. as detailed as possible, including how to handle possible edge cases):

  https://github.com/photoprism/photoprism/issues/new/choose

We can't promise at this point when exactly we'll be able to take care of it, but it's a first step!

**Reply by bitelxux** [reply]
Thank you. I'll do.

Just think that the behavior could be exactly the same that happens when pics are selected and archived from any view. The view is automatically updated and the archived pictures are removed from the view without the need of refreshing. Scroll position management could be exactly the same.

**Reply by bitelxux** [reply]
Amazing work. BTW. Congratulations and thank you very much for your effort.
